Fix the child order upshifting when parent order is deleted

Konamiman opened the pull request:
- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

#36218 implemented child order upshifting (setting their parent order id to 0) when the parent order is deleted, but this needs to
happen only when the post type of the parent order is hierarchical, otherwise child orders need to just be deleted. This pull request fixes that.

### How to test the changes in this Pull Request:

Running the unit tests should be enough, but you can also create an order with refunds, delete the order, and checking that the refund records have disappeared as well from the database.

Konamiman committed:
Fix the child order upshifting when parent order is deleted

PR 36218 implemented child order upshifting (setting their parent
order id to 0) when the parent order is deleted, but this needs to
happen only when the post type of the parent order is hierarchical,
otherwise child orders need to just be deleted. This commit fixes that.

Konamiman committed:
Add changelog file

vedanshujain commented on plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php:
```diff
@@ -1823,23 +1831,40 @@ public function delete( &$order, $args = array() ) {
+			);
+		} else {
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			$child_order_ids = $wpdb->get_col(
```
I think we'd need to call `$order->delete` here instead of immediately deleting, so that appropriate delete hooks gets fired.

Konamiman replied:
Ok, done. I had to implement the `get/set_verify_parent_id` methods in the order object too so that "Invalid parent ID" errors don't pop up.

vedanshujain replied:
unit tests you added passes for me even I remove the call to `set_verify_parent_id`, can you also add a test that captures this behavior.

vedanshujain reviewed (changes requested):
Requesting a small change, otherwise looks good.

Konamiman committed:
Delete child orders using existing functions instead of direct db access.

Additionally, add get/set_verify_parent_id methods to the order object.

vedanshujain commented on plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php:
```diff
@@ -1823,23 +1831,46 @@ public function delete( &$order, $args = array() ) {
+				$child_order->set_id( $child_order_id );
+				$child_order->delete( true );
+			}
+			// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
```
This comment should be moved to like 1861

Konamiman replied:
I must have been doing something wrong before, but now I see that I can use `wc_get_order` for the child orders and I don't get the "Invalid parent ID" errors. So I have simplified the code accordingly (including removing the new `verify_parent_id` methods).

vedanshujain reviewed (commented):
Requesting another minor change, and a test.

Konamiman committed:
Delete child orders using wc_get_order->delete

Also remove get/set_verify_parent_id methods as in the end
they don't seem to be needed.

vedanshujain reviewed (approved):
LGTM!
